pkg/migration-tool/planner/planner.go

```diff
@@ -99,11 +100,18 @@ func (c *Config) fetchLastPushedMaxt() (lastPushedMaxt int64, found bool, err er
 	if err != nil {
 		return -1, false, fmt.Errorf("create fetch-last-pushed-maxt reader: %w", err)
 	}
-	result, _, _, err := readClient.Read(context.Background(), query, "")
-	if err != nil {
+	var (
+		doneChan = make(chan *utils.PrompbResponse, 1)
+		errChan  = make(chan error, 1)
+	)
+	go readClient.Read(context.Background(), query, 0, "", doneChan, errChan)
```

Comment:
any special need to have `readClient.Read` be a goroutine now? since it spawns the concurrent routines, and here we just wait for that one routine to finish?
if there is an explicit reason why we need this to be a routine, consider using a wait group here instead of the two channels

Reply:
This is to fetch parallely. Imagine a request of 10 mins. If suppose `-concurrent-pulls=5`, then the request gets divided into:
1. first routine handles 1-2 mins.
2. second routine handles 3-4 mins.
3. third routine handles 5-6 mins.
4. ...

And after each go-routine has completed the fetching process, we combine these to form a block and sent to the writer.

The reason why I didn't use wait group is since we anyways need a channel to get the response, so why take another parameter for waitgroup. Plus, two channels acn be decreased to one by using interface channel. But this change is to be made in the writer as well, which can be done only once other PRs are merged. hence, I will do this in a different commit.

Reply:
I wouldn't use a single channel for the done and error.

And do we need readClient.Read to go in to a separate routine?

Reply:
> And do we need readClient.Read to go in to a separate routine?

How would we do parallel fetching without it?

Reply:
isn't readClient.Read the one starting the goroutines that fetch the data concurrently?

Reply:
We want the func calls (to a single function) happen concurrently, as through the same method is called, but each call does a different purpose (similar in nature i.e., fetching of data). And to make this faster, we can allow this to run parallely.

Reply:
Ah, thank you very much for pointing this out. I called this async way while dev, but it was not at all required.